R/rocRAND/build_tarballs.jl

```diff
@@ -0,0 +1,65 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "rocRAND"
+version = v"4.0.0"
+
+# Collection of sources required to build
+sources = [
+    ArchiveSource("https://github.com/ROCmSoftwarePlatform/rocRAND/archive/rocm-$(version).tar.gz",
+                  "1cafdbfa15cde635bd424d2a858dc5cc94d668f9a211ff39606ee01ed1715f41"), # 4.0.0
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/rocRAND*/
+
+amdgpu_targets="gfx900,gfx906,gfx908,gfx1010,gfx1011,gfx1012"
+
+mkdir build && cd build
+export ROCM_PATH=${prefix}
+export HIP_CLANG_PATH=${prefix}/tools
+export HIP_CLANG_HCC_COMPAT_MODE=1
+export HIP_RUNTIME=ROCclr
+export HIP_COMPILER=clang
+export HIP_PLATFORM=rocclr
+export HIPCC_VERBOSE=1
+cmake -DCMAKE_PREFIX_PATH=${prefix} \
+      -DCMAKE_INSTALL_PREFIX=${prefix} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_CXX_COMPILER=${prefix}/hip/bin/hipcc \
```

Comment:
This is not executable in the context of cross-compilation. You need to pull in a HostBuildDependency

Reply:
Thanks for the catch! `hipcc` is just a perl script, and doesn't change per-target; however, since it calls clang, that will need to be a HostBuildDependency. We still want HIP as a regular dependency because it also ships a library that we link to. In the future I might consider splitting HIP into separate JLLs.